File: lesson_20/src/main/java/lesson20/repository/ProductRepository.java
package lesson20.repository;

import lesson20.entities.Product;
import org.hibernate.SessionFactory;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class ProductRepository {
    private SessionFactory sessionFactory;
    public ProductRepository(SessionFactory sessionFactory) {
        this.sessionFactory = sessionFactory;
    }

    public void create(Product product) {
        try (var session = sessionFactory.getCurrentSession()) {
            session.getTransaction().begin();
            session.persist(product);
            session.getTransaction().commit();
        }
    }

    public Product getById(Long id) {
        Product ret = null;
        try (var session = sessionFactory.getCurrentSession()) {
            session.getTransaction().begin();
            ret = session.get(Product.class, id);
            session.getTransaction().commit();
        }
        return ret;
    }

    public List<Product> getAll() {
        List<Product> ret = null;
        try (var session = sessionFactory.getCurrentSession()) {
            session.getTransaction().begin();
            ret = session.createQuery("SELECT p FROM Product p", Product.class).getResultList();
            session.getTransaction().commit();
        }
        return ret;
    }
}

File: lesson_20/src/main/kotlin/lesson20kt/repositories/ProductRepository.kt
package lesson20kt.repositories

import lesson20kt.entities.Product
import org.hibernate.SessionFactory
import org.springframework.stereotype.Component

@Component
class ProductRepository(val sessionFactory: SessionFactory) {

    fun save(product: Product) = sessionFactory.currentSession.use { session ->
        session.transaction.begin()
        session.save(product)
        session.transaction.commit()
    }

    fun load(id: Long): Product = sessionFactory.currentSession.use { session ->
        session.transaction.begin()
        val ret = session.get(Product::class.java, id)
        session.transaction.commit()
        return ret
    }

    fun loadAll(): List<Product> = sessionFactory.currentSession.use { session ->
        session.transaction.begin()
        val ret = return session.createQuery("SELECT p FROM Product p", Product::class.java).list()
        session.transaction.commit()
        return ret
    }
}